### PR TITLE
[#2721] Update paver from python-package

### PR DESCRIPTION
**Pull request by dumol**
Problem?
----------
An older version of `paver.sh` is being updated in python-package for ARM64 and HP-UX support.

Solution?
----------
Forward-patch the improvements from python-package to brink, so that we may update `paver.sh` in python-package from brink without losing latest changes.

**Drive-by fix**:
  * use a better default value for `$flag_supported`.

How to check?
-----------------
Please review changes.
Check `bash -x paver.sh os_detect` in HP-UX and Ubuntu 14.04 on ARM.
Run tests.

reviewer: @adiroiban

### CODE REVIEW

**Comment by dumol** [comment]
**needs-review**

**Comment by adiroiban** [comment]
All good. thanks!

hope that soon we will have an hpux slave and we can test the code there.

approved-at 4efdf12

**Comment by dumol** [comment]
Just for fun, have looked-up free shell accounts for HP-UX and run into this: http://www.polarhome.com

Might be interesting to consider.

**Comment by dumol** [comment]
The version in `setup.py` should have been updated too. Hard-reseted HEAD and trying to re-open now…

**Comment by dumol** [comment]
**needs-review**

**Comment by dumol** [comment]
And a question: Shouldn't we use `hpux1131` for naming the package? Currently I see you have only used `hpux` for the OS part in the package name, which is not in line with the other names: `solaris10`, `rhel4`, `sles11`, `ubuntu1204` etc.

File: paver.sh [hunk 1]
```diff
@@ -427,9 +427,13 @@ detect_os() {
...
     elif [ "${OS}" = "hp-ux" ]; then
 
+        # libffi and GMP do not compile with the HP compiler, so we use GCC.
+
         ARCH=$(uname -m)
+        os_version_raw=$(uname -r | cut -d'.' -f2-)
+        check_os_version HP-UX 11.31 "$os_version_raw" os_version_fancy
 
-        OS="hpux"
+        OS="hpux${os_version_fancy}"
```
Comment (adiroiban) [inline paver.sh]: what is os_version_fancy ?
while the name might be  cute or clever I have no idea what is should hold.

Reading the documentation of check_os_version I see that is the OS name used by our binary packages

`os_binary_name` or `os_fingerprint` or something like that should be a better name than `fancy`.

**Comment by adiroiban** [comment]
Change looks good.

for hpux we should have hpux1131 

approved-at latest

Thanks!

File: paver.sh (hunk 1 above)
Comment (dumol) [inline paver.sh]: Fine by me, as discussed on IRC this is going to be `os_version_chevah`.

**Pull request closed without merging**